src/DepartmentService.Broker/FilterDepartmentsUsersConsumer.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LT.DigitalOffice.DepartmentService.Data.Interfaces;
+using LT.DigitalOffice.DepartmentService.Models.Db;
+using LT.DigitalOffice.Kernel.BrokerSupport.Broker;
+using LT.DigitalOffice.Models.Broker.Models.Department;
+using LT.DigitalOffice.Models.Broker.Requests.Department;
+using LT.DigitalOffice.Models.Broker.Responses.Department;
+using MassTransit;
+
+namespace LT.DigitalOffice.DepartmentService.Broker
+{
+  public class FilterDepartmentsUsersConsumer : IConsumer<IFilterDepartmentsRequest>
+  {
+    private readonly IDepartmentRepository _repository;
+
+    public async Task<List<DepartmentFilteredData>> GetDepartmentFilteredData(IFilterDepartmentsRequest request)
+    {
+      List<DbDepartment> dbPosition = await _repository.GetDepartmenDataAsync(request.DepartmentsIds);
+
+      return dbPosition.Select(
+        pd => new DepartmentFilteredData(
+          pd.Id,
+          pd.Name,
+          pd.Users.Where(u => u.IsActive).Select(u => u.UserId).ToList())).ToList();
```

Comment:
no need select IsActiveUsers, you done it in repo

Reply:
fixed